### ISSUE TITLE
Unzipping progress goes to 200% 

### ISSUE THREAD

**Issue by goodliff** [cypress-io/cypress#5967]
Tried the last release (3.8.0) on docker to check the the unzipping change that was introduced by cli: unnzip Cypress using `unzip` utility on Linux #5851, when downloading and extracting cypress I noticed that the unzipping progress continued to 200% and time remaining went into negative figures. 

### Current behavior:
```
[14:09:00] Unzipping Cypress 200% -56s [title changed]
```

### Desired behavior:

Ideally this would be accurate to the unzip progress.

### Versions

Cypress package version: 3.8.0 | Cypress binary version: 3.8.0 | docker-compose version 1.24.1, build 4667896b

**Comment by GregLahaye** [comment]
Experienced same issue with Cypress 3.8.0 on Windows 10 WSL Ubuntu

**Comment by jennifer-shehane** [comment]
😅 Yeah, that should not be happening. We have some tests around this, but no tests around the exact percentage display as far as I can tell.

Can you run the install with in [debug mode](https://on.cypress.io/debugging#Debug-the-Command-Line) mode and print the entire set of logs here? 

There are 2 pathways it could take for unzipping and this should tell us which pathway yours followed.

**Comment by GregLahaye** [comment]
Here is debug log from Windows 10 WSL Ubuntu
https://pastebin.com/aqWyJ0aX

**Comment by goodliff** [comment]
Greg I'm a bit slow, did that take 14 minutes to unzip? I'm struggling to convince people to get it into our build cycles with a 4 minute (haven't checked the time since 3.8.0) unzip time. I'm assuming you're able to cache it between builds?

**Comment by GregLahaye** [comment]
😄 yea it is a long time - this was on WSL (file I/O takes longer) and also yes - it will cache after first install so you should only need to install once (per version at least).

**Comment by flotwig** [comment]
@GregLahaye @goodliff It does still unzip correctly, right? Just the progress indicator is wrong?

**Comment by flotwig** [comment]
If you guys can do me a favor and send me the output of `unzip -v` - it would be helpful.

The progress indicator works basically by counting the number of lines output and dividing that by the total number of file in the archive - it assumes 1 line per file output. But maybe there is some version of unzip that has different stdout conventions.

**Comment by goodliff** [comment]
Wow! Have you sped that up for demonstration purposes? :) 

I'm just in the middle of something but when I'm free I'll try and get more information from docker (had to remove it from dependencies for the moment)

**Comment by flotwig** [comment]
Nope, it should be this fast if you have `unzip` installed! That was supposed to be an improvement in 3.8.0

re: `unzip` info - Sounds good - lemme know. It's also possible that your system doesn't have `unzip`, which could be causing the regression.

**Comment by goodliff** [comment]
Quite possibly, it's running via docker-compose at the moment and I've tried adding unzip to that. I think it may be defaulting to the npm unzip, I believe we have that as a dev-dep. 

```
> cypress@3.8.0 postinstall /usr/src/app/node_modules/cypress
> node index.js --exec install

Installing Cypress (version: 3.8.0)

[16:41:16]  Downloading Cypress     [started]
[16:41:16]  Downloading Cypress      0% 0s [title changed]
[16:41:16]  Downloading Cypress      1% 20s [title changed]
[16:41:17]  Downloading Cypress      1% 30s [title changed]
[16:41:17]  Downloading Cypress      1% 40s [title changed]
...
[16:41:59]  Downloading Cypress      96% 2s [title changed]
[16:41:59]  Downloading Cypress      97% 1s [title changed]
[16:42:00]  Downloading Cypress      98% 1s [title changed]
[16:42:00]  Downloading Cypress      99% 0s [title changed]
[16:42:00]  Downloading Cypress      100% 0s [title changed]
[16:42:01]  Downloaded Cypress      [title changed]
[16:42:01]  Downloaded Cypress      [completed]
[16:42:01]  Unzipping Cypress       [started]
[16:42:02]  Unzipping Cypress        0% 0s [title changed]
[16:42:08]  Unzipping Cypress        1% 684s [title changed]
[16:42:08]  Unzipping Cypress        1% 685s [title changed]
[16:42:08]  Unzipping Cypress        1% 686s [title changed]
...
[16:43:33]  Unzipping Cypress        97% 3s [title changed]
[16:43:34]  Unzipping Cypress        98% 2s [title changed]
[16:43:35]  Unzipping Cypress        99% 1s [title changed]
[16:43:35]  Unzipping Cypress        100% 0s [title changed]
[16:43:36]  Unzipping Cypress        101% -1s [title changed]
[16:43:36]  Unzipping Cypress        102% -2s [title changed]
[16:43:38]  Unzipping Cypress        103% -3s [title changed]
[16:43:38]  Unzipping Cypress        104% -4s [title changed]
...
[16:44:38]  Unzipping Cypress        198% -78s [title changed]
[16:44:39]  Unzipping Cypress        199% -78s [title changed]
[16:44:39]  Unzipped Cypress        [title changed]
[16:44:39]  Unzipped Cypress        [completed]
[16:44:39]  Finishing Installation  [started]
[16:44:39]  Finished Installation   /root/.cache/Cypress/3.8.0 [title changed]
[16:44:39]  Finished Installation   /root/.cache/Cypress/3.8.0 [completed]

You can now open Cypress by running: node_modules/.bin/cypress open

https://on.cypress.io/installing-cypress
```

I've chopped large bits out of this to make it more readable, I'm still working on getting -v on unzip but I'll have to rebuild which will take some time.

Off topic, I'm trying to workout whether a PR from 8 months ago got released, where is the best place to ask or should I just raise a ticket?

**Comment by goodliff** [comment]
Yep, looks like you're right, I installed the unzip via apt before the installation and it completed in 20 secs instead of the few minutes without it. It also calculated the progress correctly. 

```[17:17:48]  Unzipping Cypress        44% 12s [title changed]
[17:17:48]  Unzipping Cypress        45% 12s [title changed]
[17:17:48]  Unzipping Cypress        46% 11s [title changed]
[17:17:48]  Unzipping Cypress        47% 11s [title changed]
[17:17:48]  Unzipping Cypress        48% 11s [title changed]
[17:17:48]  Unzipping Cypress        49% 10s [title changed]
[17:17:48]  Unzipping Cypress        50% 10s [title changed]
[17:17:49]  Unzipping Cypress        51% 10s [title changed]
[17:17:49]  Unzipping Cypress        52% 9s [title changed]
[17:17:49]  Unzipping Cypress        52% 10s [title changed]
[17:17:49]  Unzipping Cypress        100% 0s [title changed]
[17:17:49]  Unzipped Cypress        [title changed]
[17:17:49]  Unzipped Cypress        [completed]
[17:17:49]  Finishing Installation  [started]
[17:17:49]  Finished Installation   /root/.cache/Cypress/3.8.0 [title changed]
[17:17:49]  Finished Installation   /root/.cache/Cypress/3.8.0 [completed]

You can now open Cypress by running: node_modules/.bin/cypress open```

**Comment by GregLahaye** [comment]
Same thing here - didn't have `unzip` installed so it seems like Cypress fell back to unzipping with Node.js:

2019-12-17T10:57:23.765Z cypress:cli:unzip unzipping from /tmp/cypress.zip
2019-12-17T10:57:23.766Z cypress:cli:unzip into /home/greg/.cache/Cypress/3.8.0
2019-12-17T10:57:23.785Z cypress:cli:unzip zipFile entries count 39363
2019-12-17T10:57:23.786Z cypress:cli:unzip unzipping via `unzip`
2019-12-17T10:57:23.868Z cypress:cli:unzip unzipping with node.js (slow)
2019-12-17T10:57:23.884Z cypress:cli:unzip `unzip` failed { code: -2 }
2019-12-17T10:57:23.884Z cypress:cli:unzip unzipping with node.js (slow)

Installing unzip through apt fixes the issue.

**Comment by goodliff** [comment]
Need anything else @flotwig?

**Comment by flotwig** [comment]
@goodliff Nope, we're good, thanks for the details so far. I've reproduced it locally and I have a PR open to fix it here: #6009

**Comment by jennifer-shehane** [comment]
Released in [3.8.2](https://on.cypress.io/changelog#3-8-2)
